### ISSUE TITLE
Does not support Android

### ISSUE THREAD

**Issue by spentak** [nemtech/nem2-sdk-java#8]
Many of the dependencies do not support any version of Java that Android uses.

Was this by design?

I don't see any packages that needed to be on the latest Java, seems like it was possibly an oversight.

How should we proceed supporting Android? Meaning, we as open source community.

**Comment by aleixmorgadas** [comment]
It's not desired, the first version was made for server side, and fully support for Android was planned for later versions.

We need to change some libraries to be more lightweight for Android regarding for the internal implementations.

Have you an Android project ongoing that requires Android support asap?

**Comment by aleixmorgadas** [comment]
> I think mobile is critical for adoption

Yes, we agree that mobile is critical for adoption, no discussion here ^_^

Seems like the incompatible libraries are the logger and the vertx, but might be more. Hopefully, this just requires change the underlying implementation instead the public API.

For now, I just might drop that implementation and replace it with Android compatibles.

Replacing the Vertx by OkHttp and drop the logging if necessary should solve the problem.

**Comment by robotictang** [comment]
Hi all, anyone is doing the "Replacing the Vertx by OkHttp and drop the logging" for android compatibility?

**Comment by ryuta46** [comment]
> anyone is doing the "Replacing the Vertx by OkHttp and drop the logging" for android compatibility?

I tried in my forked repository.
https://github.com/ryuta46/nem2-sdk-java

I found that not only logging library and HTTP client(Vert.x), libraries below are not worked on Android.

* Vert.x
* log4j
* commons-codec
* BouncyCastle
* java.time package in Java 8

I replaced them  in the above repository. I will send pull request in a week.

**Comment by fboucquez** [comment]
Hi @spentak, @ryuta46, @robotictang 

We have released a new version of the SDK, 0.14.0. This new SDK comes in 2 flavors OkHttp and Vertx.

https://repo.maven.apache.org/maven2/io/nem/sdk-vertx-client/0.14.0/
https://repo.maven.apache.org/maven2/io/nem/sdk-okhttp-client/0.14.0/

We haven't focused on all the problems related to Android and the libs problem you have mentioned yet (bit thanks for that) but we want to help you create Android apps soon. 

If you want to try the new OkHttp client in Android, please use it. Removing Vertx from the core work was a big step towards Android compatibility.

I believe the only problems remaining are 

- commons-codec
- BouncyCastle
- java.time package in Java 8

if you find any other problem, please report it here. 

Much appreciated 

Fernando
